File: server/src/main/java/au/com/codeka/warworlds/server/admin/handlers/AjaxEmpireHandler.java
package au.com.codeka.warworlds.server.admin.handlers;

import au.com.codeka.warworlds.common.proto.Empire;
import au.com.codeka.warworlds.server.handlers.RequestException;
import au.com.codeka.warworlds.server.world.EmpireManager;
import au.com.codeka.warworlds.server.world.WatchableObject;

/** Handler for /admin/ajax/empires/&lt;id&gt; */
public class AjaxEmpireHandler extends AdminHandler {
  @Override
  public void get() throws RequestException {
    if (getRequest().getParameter("id") != null) {
      long empireId = Long.parseLong(getRequest().getParameter("id"));
      WatchableObject<Empire> empire = EmpireManager.i.getEmpire(empireId);
      if (empire == null) {
        throw new RequestException(404);
      }
      setResponseJson(empire.get());
    }
  }
}

File: server/src/main/kotlin/au/com/codeka/warworlds/server/admin/handlers/AjaxEmpireHandler.kt
package au.com.codeka.warworlds.server.admin.handlers

import au.com.codeka.warworlds.server.handlers.RequestException
import au.com.codeka.warworlds.server.world.EmpireManager

/** Handler for /admin/ajax/empires/&lt;id&gt;  */
class AjaxEmpireHandler : AdminHandler() {
  public override fun get() {
    if (request.getParameter("id") != null) {
      val empireId = request.getParameter("id").toLong()
      val empire = EmpireManager.i.getEmpire(empireId) ?: throw RequestException(404)
      setResponseJson(empire.get())
    }
  }
}
